src/wfs/src/main/java/org/geoserver/wfs/json/GeoJSONBuilder.java

```diff
@@ -286,19 +288,65 @@ public static int getGeometryType(Geometry geometry) {
                 "Unable to determine geometry type " + geometry.getClass());
         }
     }
-    
+
+    /**
+     * Write a java.util.List out as a JSON Array. The values of the array will be converted
+     * using ike standard primitive conversions. If the list contains List or Map objects, they
+     * will be serialized as JSON Arrays and JSON Objects respectively.
+     *
+     * @param list a java.util.List to be serialized as JSON Array
+     */
+    public JSONBuilder writeList(final List list) {
+        this.array();
+        for (final Object o: list) {
+            this.value(o);
+        }
+        return this.endArray();
+    }
+
     /**
-     * Overrides to handle the case of encoding {@code java.util.Date} and its date/time/timestamp
+     * Write a java.util.Map out as a JSON Object. Keys are serialized using the toString method
+     * of the object and values are serialized using primitives conversions. If a value in the map
+     * is a List or Map object, it will be serialized as JSON Array or JSON Object respectively.
+     *
+     * @param map a java.util.Map object to be serialized as a JSON Object
+     */
+    public JSONBuilder writeMap(final Map map) {
+        this.object();
+        for (final Object k: map.keySet()) {
+            this.key(k.toString());
+            this.value(map.get(k));
+        }
+        return this.endObject();
+    }
+
+    /**
+     * Overrides handling of specialized types.
+     *
+     * Overrides the encoding {@code java.util.Date} and its date/time/timestamp
      * descendants, as well as {@code java.util.Calendar} instances as ISO 8601 strings.
-     * 
+     *
+     * Overrides the handling of java.util.Map, java.util.List, and Geometry objects
+     * as well.
+     *
      * @see net.sf.json.util.JSONBuilder#value(java.lang.Object)
      */
     @Override
     public GeoJSONBuilder value(Object value) {
-        if (value instanceof java.util.Date || value instanceof Calendar) {
-            value = Converters.convert(value, String.class);
+        if (value == null) {
+            super.value(value);
+        } else if (value instanceof Geometry) {
+            this.writeGeom((Geometry) value);
```

Comment:
Was writeGeom not being used before? May want to check if it uses the correct "curve aware" geotools wkt writer (rather than just calling geometry toString() ).